pilot/pkg/xds/ecds.go

```diff
@@ -71,5 +117,127 @@ func (e *EcdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w
 			Resource: util.MessageToAny(c),
 		})
 	}
+
 	return resources, model.DefaultXdsLogDetails, nil
 }
+
+func (e *EcdsGenerator) GeneratePullSecrets(proxy *model.Proxy, updatedSecrets map[model.ConfigKey]struct{}, secretResources []SecretResource,
+	secretController credscontroller.Controller, req *model.PushRequest) map[string][]byte {
+	if proxy.VerifiedIdentity == nil {
+		log.Warnf("proxy %s is not authorized to receive secret. Ensure you are connecting over TLS port and are authenticated.", proxy.ID)
+		return nil
+	}
+
+	results := make(map[string][]byte)
+	for _, sr := range secretResources {
+		if updatedSecrets != nil {
+			if _, found := updatedSecrets[model.ConfigKey{Kind: gvk.Secret, Name: sr.Name, Namespace: sr.Namespace}]; !found {
+				// This is an incremental update, filter out credscontroller that are not updated.
+				continue
+			}
+		}
+
+		if cachedItem, f := e.secretCache.Get(sr); f {
+			// If it is in the Cache, add it and continue
+			sb, err := toSecretBytes(cachedItem)
```

Comment:
It seems the cache does not help for performance, and it adds conversion

Reply:
I thought it will reduce API list call for secret, although it will be indeed meaningless if credential controller already have a cache. Not sure if that is case?

Reply:
I think credential store will have informer cache

Reply:
Ok, removed the cache here. thanks!